yt_dlp/extractor/loco.py

```diff
@@ -0,0 +1,82 @@
+from .common import InfoExtractor
+from ..utils import traverse_obj
```

Comment:
try to be concise with import location, ie
```suggestion
from ..utils.traversal import traverse_obj
```

Reply:
IMO this is just creates a tighter and less maintainable binding between `utils` and extractors that depend on it. Why should the extractor need to know how functions in the `utils` module/package are implemented? Suppose someone takes over maintenance and wants to put `traverse_obj` in `utils.traversal_ext_2_pro`: then every extractor that mentions `utils.traversal` has to be tweaked, or the function has to be re-exported from `utils.traversal`, when re-exporting it from `utils.__init__ already avoids the problem. Or am I missing something?

The same point could apply to `HEADRequest`.

Reply:
> try to be concise with import location

@Grub4K Since I'm adding `ExtractorError`, this would require to split the imports. Is it better to use a single import from `..utils` or split it in an import from `..utils` and one from `..utils.traversal`?

Reply:
Its probably better to use `traversal.require` instead, that seems cleaner to me. In the case you need both locations, you would usually split it into two imports:
```py
from ..utils import ExtractorError
from ..utils.traversal import traverse_obj
```

Reply:
While the ship for module separation has sailed quite some time ago, I still want to push it back into that direction if possible. We reexport the functions from `utils` ONLY for compatibility. Imports should always be as precise as possible so that import resolving can prune unnecessary functions. `utils` is overloaded and I would accept a combined import only if the reexport were lazy (which I had already prototyped for utils module split before).

> The same point could apply to `HEADRequest`.

And I think the same *should* apply for `HEADRequest`. If I see it now I will complain, thanks!